src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // constructor(name) {
+  //   super(name);
+  // }
+  bite(animal) {
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    if (animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        const idx = Animal.alive.indexOf(animal);
+
+        Animal.alive.splice(idx, 1);
+      }
+    }
```

Comment:
The logic checks if the `animal` is not hidden before reducing its health. Ensure this behavior is consistent with the task requirements. If hidden animals should still be affected, this condition needs to be revised.